File: codingame/community/JackSilverTheCasinoPart1/JackSilverTHeCasino1.php
<?php
fscanf(STDIN, "%d",
    $rnds
);
fscanf(STDIN, "%d",
    $cash
);
#echo("$rnds\n");
#echo("$cash\n");

for ($i = 0; $i < $rnds; $i++)
{
    $play = stream_get_line(STDIN, 1024 + 1, "\n");
    $plays = explode(" ", $play);
    
    $bet = ceil($cash/4);
#    echo("bet: $bet\n");
    $cash = $cash - $bet;
#    echo("cash: $cash\n");
    $ball = intval($plays[0]);
    $call = $plays[1];
    
    if ($call == "PLAIN") {
        $target = intval($plays[2]);
    }
#    echo("$call : $ball - $target\n");
    if ($call == "PLAIN" && $ball == $target) {
        $cash = $cash + ($bet * 36);
    }elseif ($call == "EVEN" && $ball > 0 && $ball % 2 == 0) {
        $cash = $cash + ($bet * 2);
    }elseif ($call == "ODD" && $ball % 2 == 1) {
        $cash = $cash + ($bet * 2);
    }
#    echo("new cash: $cash\n");
#    echo("-----------\n");
}

echo("$cash\n");
?>

File: codingame/community/JackSilverTheCasinoPart1/JackSilverTHeCasino1.dart
import 'dart:io';
import 'dart:math';

void main() {
    int rnds = int.parse(stdin.readLineSync());
    int cash = int.parse(stdin.readLineSync());
    for (int i = 0; i < rnds; i++) {
        String play = stdin.readLineSync();
    
        int bet = (cash/4).ceil().toInt();
        cash = cash - bet;
        List<string> plays = play.split(' ');
        int ball = int.parse(plays[0]);
        String call = plays[1];
        int target;
        if (call == "PLAIN"){
            target = int.parse(plays[2]);
        }
        
        if (call == "PLAIN" && ball == target) {
            cash = cash + (bet * 36);
        }else if (call == "EVEN" && ball > 0 && ball % 2 == 0) {
            cash = cash + (bet * 2);
        }else if (call == "ODD" && ball % 2 == 1) {
            cash = cash + (bet * 2);
        }
    }
    print(cash);
}